README.md

```diff
@@ -112,3 +121,138 @@ make
 # T2 Hardware API
 
 When you `require('tessel')` within a script which is executed on Tessel 2, this loads a library which interfaces with the Tessel 2 hardware, including pins, ports, and LEDs, just like Tessel 1 ([Tessel 1 hardware documentation](https://tessel.io/docs/hardwareAPI)). The code for Tessel 2's hardware object can be found [here](https://github.com/tessel/t2-firmware/blob/master/node/tessel.js).
+
+## Ports and pins
+
+Tessel has two ports, A and B. They are referred to as `tessel.port['A']`.
+
+Tessel's ports can be used as module ports as in Tessel 1 (e.g. `accelerometer.use(tessel.port['B'])`), or used as flexible GPIO pins (e.g. `myPin = tessel.port['A'].pins[0]`).
+
+### Modules
+
+Tessel 2's module ports can be used with [Tessel modules](//tessel.io/modules) much as in [Tessel 1](http://start.tessel.io/modules).
+
+Here is an example of using the Tessel Climate module on Tessel's port B:
+
+```js
+var tessel = require('tessel');
+var climatelib = require('climate-si7020');
+var climate = climatelib.use(tessel.port['B']);
+```
+
+### Pin mapping
+
+The module ports are not just for modules! They can also be used as flexible, simply addressable GPIO pins.
+
+The pin layout for ports A and B is as follows:
+
+| Index | Function |
+|-------|----------|
+| 0     | SCL      |
+| 1     | SDA      |
+| 2     | SCK      |
+| 3     | MISO     |
+| 4     | MOSI     |
+| 5     | TX       |
+| 6     | RX       |
+| 7     | I/O      |
+
+All pins (0-7) can be used as digital pins. All pins on port B and pin 7 on port A can be used as analog pins.
+
```

Comment:
Recommend combining this information with the table above into a single table: 

| Port | Pin | Capabilities |
|------|-----|----------|
|A     | 0   | Digital I/O, SCL |
|A     | 1   | Digital I/O, SDA |
|A     | 2   | Digital I/O, SCK |
|A     | 3   | Digital I/O, MISO |
|A     | 4   | Digital I/O, MOSI, Analog Input |
|A     | 5   | Digital I/O, TX |
|A     | 6   | Digital I/O, RX |
|A     | 7   | Digital I/O, Analog Input |
|B     | 0   | Digital I/O, SCL, Analog Input |
|B     | 1   | Digital I/O, SDA, Analog Input |
|B     | 2   | Digital I/O, SCK, Analog Input |
|B     | 3   | Digital I/O, MISO, Analog Input |
|B     | 4   | Digital I/O, MOSI, Analog Input |
|B     | 5   | Digital I/O, TX, Analog Input |
|B     | 6   | Digital I/O, RX, Analog Input |
|B     | 7   | Digital I/O, Analog Input, Analog Output |


Or


| Port | Pin | Digital I/O | SCL | SDA | SCK | MISO | MOSI | TX | RX | Analog In | Analog Out |
|------|-----|-------------|-----|-----|-----|------|------|----|----|-----------|------------|
|A     | 0   | ✓           | ✓   |     |     |      |      |    |    |           |            |
|A     | 1   | ✓           |     | ✓   |     |      |      |    |    |           |            |
|A     | 2   | ✓           |     |     | ✓   |      |      |    |    |           |            |
|A     | 3   | ✓           |     |     |     | ✓    |      |    |    |           |            |
|A     | 4   | ✓           |     |     |     |      | ✓    |    |    | ✓         |            |
|A     | 5   | ✓           |     |     |     |      |      | ✓  |    |           |            |
|A     | 6   | ✓           |     |     |     |      |      |    | ✓  |           |            |
|A     | 7   | ✓           |     |     |     |      |      |    |    | ✓         |            |
|B     | 0   | ✓           | ✓   |     |     |      |      |    |    | ✓         |            |
|B     | 1   | ✓           |     | ✓   |     |      |      |    |    | ✓         |            |
|B     | 2   | ✓           |     |     | ✓   |      |      |    |    | ✓         |            |
|B     | 3   | ✓           |     |     |     | ✓    |      |    |    | ✓         |            |
|B     | 4   | ✓           |     |     |     |      | ✓    |    |    | ✓         |            |
|B     | 5   | ✓           |     |     |     |      |      | ✓  |    | ✓         |            |
|B     | 6   | ✓           |     |     |     |      |      |    | ✓  | ✓         |            |
|B     | 7   | ✓           |     |     |     |      |      |    |    | ✓         | ✓          |